cirq/google/engine/engine_job.py

```diff
@@ -39,53 +41,182 @@ class EngineJob:
+        return self
+
+    def add_labels(self, labels: Dict[str, str]) -> 'EngineJob':
+        """Adds new labels to a previously created quantum job.
+
+        Params:
+            labels: New labels to add to the existing job labels.
+
+        Returns:
+             This EngineJob.
+        """
+        self._job = self.context.client.add_job_labels(self.project_id,
+                                                       self.program_id,
+                                                       self.job_id, labels)
+        return self
+
+    def remove_labels(self, keys: List[str]) -> 'EngineJob':
+        """Removes labels with given keys from the labels of a previously
+        created quantum job.
+
+        Params:
+            label_keys: Label keys to remove from the existing job labels.
+
+        Returns:
+            This EngineJob.
+        """
+        self._job = self.context.client.remove_job_labels(
+            self.project_id, self.program_id, self.job_id, keys)
+        return self
+
     def status(self) -> str:
         """Return the execution status of the job."""
         return quantum.types.ExecutionStatus.State.Name(
             self._refresh_job().execution_status.state)
 
+    def get_repetitions_and_sweeps(self) -> Tuple[int, List[study.Sweep]]:
+        """Returns the repetitions and sweeps for the Quantum Engine job.
+
+        Returns:
+            A tuple of the repetition count and list of sweeps.
+        """
+        if not self._job.HasField('run_context'):
+            self._job = self.context.client.get_job(self.project_id,
+                                                    self.program_id,
+                                                    self.job_id, True)
+
+        return self._deserialize_run_context(self._job.run_context)
+
+    @staticmethod
+    def _deserialize_run_context(run_context: quantum.types.any_pb2.Any
+                                ) -> Tuple[int, List[study.Sweep]]:
+        import cirq.google.engine.engine as engine_base
+        run_context_type = run_context.type_url[len(engine_base.TYPE_PREFIX):]
+        if (run_context_type == 'cirq.google.api.v1.RunContext' or
+                run_context_type == 'cirq.api.google.v1.RunContext'):
+            raise ValueError('deserializing a v1 RunContext is not supported')
+        if (run_context_type == 'cirq.google.api.v2.RunContext' or
+                run_context_type == 'cirq.api.google.v2.RunContext'):
+            v2_run_context = v2.run_context_pb2.RunContext()
+            v2_run_context.ParseFromString(run_context.value)
+            return v2_run_context.parameter_sweeps[0].repetitions, [
+                v2.sweep_from_proto(s.sweep)
+                for s in v2_run_context.parameter_sweeps
+            ]
+        raise ValueError(
+            'unsupported run_context type: {}'.format(run_context_type))
+
     def get_calibration(self) -> Optional[calibration.Calibration]:
```

Comment:
Should we also have a get_processor? There should at least be a way to get the processor_id from the job.